.github/workflows/cypress.yml

```diff
@@ -33,36 +42,51 @@ jobs:
       - name: Install bootstrapper dependencies
         run: pipenv install --dev --deploy
       - run: pipenv run cookiecutter . --no-input -f
+      - uses: actions/upload-artifact@v3
+        with:
+          name: my_project
+          path: my_project/
+          retention-days: 1
+    outputs:
+      BASE_URL: ${{ env.BASE_URL }}
+  Chrome:
+    needs: Setup
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - uses: actions/download-artifact@v3
+        with:
+          name: my_project
+          path: my_project/
       - name: Install frontend dependencies
         env:
           NPM_CONFIG_PRODUCTION: false
         working-directory: ./my_project/client
         run: npm install
```

Comment:
But for some reason `npm install` during `Setup` wouldn't work. Didn't bother digging into why.

Reply:
If it was a react app then it could be due to that `Apollo` peer dependencies error that is also behind the ESLint tests failing

Reply:
It was on a Vue app. I think the error was something about the lock file missing. Something not getting carried over during the upload/download artifact step. I'm not really worried, though I like the idea of doing all the build steps during setup only.

Reply:
I understand this better now. Makes perfect sense now that I think about.
Storing artifacts via `actions/upload-artifact` is only for files and folders. For things like installs via apt/python/npm there is a separate `caches` github action (`actions/cache`) which is also available inside actions like `actions/setup-python` (which we use). That is the proper way to install something once, and then reuse it between jobs or even separate workflow runs.
You can in fact install something when running the `main` branch and it'll use that cache by default on PRs.
artifacts are less powerful and are attached to a specific workflow run. So they can't be reused more widely.